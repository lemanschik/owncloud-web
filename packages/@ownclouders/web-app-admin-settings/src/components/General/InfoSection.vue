<template>
  <div>
    <h2 class="oc-py-s" v-text="$gettext('Info')" />

    <table class="details-table">
      <tr>
        <th scope="col" class="oc-pr-s" v-text="$gettext('ownCloud')" />
        <td v-text="backendProductName" />
      </tr>
      <tr>
        <th scope="col" class="oc-pr-s" v-text="$gettext('Edition')" />
        <td v-text="backendEdition" />
      </tr>
      <tr>
        <th scope="col" class="oc-pr-s" v-text="$gettext('Version')" />
        <td v-text="backendVersion" />
      </tr>
      <tr>
        <th scope="col" class="oc-pr-s" v-text="$gettext('Web client version')" />
        <td v-text="webClientVersion" />
      </tr>
    </table>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { useStore } from 'web-pkg/src/composables'

export default defineComponent({
  name: 'InfoSection',
  setup() {
    const store = useStore()
    let backendProductName = ''
    let backendVersion = ''
    let backendEdition = ''
    let webClientVersion = ''

    const backendStatus = store.getters.capabilities?.core?.status

    if (backendStatus && backendStatus.versionstring) {
      backendProductName = backendStatus.product || 'ownCloud'
      backendVersion = backendStatus.productversion || backendStatus.versionstring
      backendEdition = backendStatus.edition
      webClientVersion = process.env.PACKAGE_VERSION
    }

    return {
      webClientVersion,
      backendProductName,
      backendVersion,
      backendEdition
    }
  }
})
</script>
