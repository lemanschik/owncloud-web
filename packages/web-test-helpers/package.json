{
  "name": "web-test-helpers",
  "private": true,
  "main": "src/index.ts",
  "peerDependencies": {
    "@casl/ability": "^6.3.3",
    "@casl/vue": "^2.2.1",
    "axios": "0.27.2",
    "design-system": "@ownclouders/design-system",
    "vue3-gettext": "^2.3.3",
    "vue-router": "4.1.6",
    "vuex": "4.1.0"
  }
}
