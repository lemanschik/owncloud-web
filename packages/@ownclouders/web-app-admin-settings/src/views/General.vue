<template>
  <div>
    <app-template ref="template" :breadcrumbs="breadcrumbs">
      <template #mainContent>
        <div class="oc-px-m">
          <InfoSection />
          <AppearanceSection />
        </div>
      </template>
    </app-template>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref } from 'vue'
import AppTemplate from '../components/AppTemplate.vue'
import InfoSection from '../components/General/InfoSection.vue'
import AppearanceSection from '../components/General/AppearanceSection.vue'

export default defineComponent({
  components: {
    AppearanceSection,
    AppTemplate,
    InfoSection
  },
  setup() {
    const template = ref()

    return {
      template
    }
  },
  computed: {
    breadcrumbs() {
      return [
        { text: this.$gettext('Administration Settings'), to: { path: '/admin-settings' } },
        {
          text: this.$gettext('General')
        }
      ]
    }
  }
})
</script>
