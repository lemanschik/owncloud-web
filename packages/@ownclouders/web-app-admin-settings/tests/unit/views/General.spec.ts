import General from '../../../src/views/General.vue'
import { defaultPlugins, shallowMount } from 'web-test-helpers'

describe('General view', () => {
  it('renders component', () => {
    const { wrapper } = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
})

function getWrapper() {
  return {
    wrapper: shallowMount(General, {
      global: {
        plugins: [...defaultPlugins()]
      }
    })
  }
}
