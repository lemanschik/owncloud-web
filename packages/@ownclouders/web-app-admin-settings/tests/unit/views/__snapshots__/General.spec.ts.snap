// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General view renders component 1`] = `
<div>
  <app-template-stub batchactionitems="" batchactions="" breadcrumbs="[object Object],[object Object]" issidebarheadercompact="false" loading="false" showbatchactions="false" sidebaravailablepanels="" sidebarloading="false" sidebaropen="false"></app-template-stub>
</div>
`;
